backend/src/schema/resolvers/fileUpload/index.js

```diff
@@ -1,24 +1,27 @@
 import { createWriteStream } from 'fs'
 import path from 'path'
 import slug from 'slug'
+import uuid from 'uuid/v4'
 
-const storeUpload = ({ createReadStream, fileLocation }) =>
-  new Promise((resolve, reject) =>
+const localFileUpload = async ({ createReadStream, uniqueFilename }) => {
+  await new Promise((resolve, reject) =>
     createReadStream()
-      .pipe(createWriteStream(`public${fileLocation}`))
+      .pipe(createWriteStream(`public${uniqueFilename}`))
       .on('finish', resolve)
       .on('error', reject),
   )
+  return uniqueFilename
+}
 
-export default async function fileUpload(params, { file, url }, uploadCallback = storeUpload) {
+export default async function fileUpload(params, { file, url }, uploadCallback = localFileUpload) {
   const upload = params[file]
   if (upload) {
     const { createReadStream, filename } = await upload
-    const { name } = path.parse(filename)
-    const fileLocation = `/uploads/${Date.now()}-${slug(name)}`
-    await uploadCallback({ createReadStream, fileLocation })
+    const { name, ext } = path.parse(filename)
+    const uniqueFilename = `/uploads/${uuid()}-${slug(name)}${ext}`
```

Comment:
@mattwr18 would comment "code creep" here.

Reply:
yeah, I thought of that, but the challenge I was facing is that we can't cypress test if we call `canvas.toBlob` or at least I couldn't find a workaround. Until I remembered that we wanted to stop compressing/converting `png/svg`.
This is a great example of an instance where if we had used `TDD`, we would have implemented the feature in a way that dealt with bugs we now have